api/schema_system.go

```diff
@@ -0,0 +1,75 @@
+
+// SchemaReviewRuleLevel is the error level for schema review rule.
+type SchemaReviewRuleLevel string
+
+// SchemaReviewRuleType is the type of schema rule.
+type SchemaReviewRuleType string
+
+const (
+	// SchemaRuleLevelError is the error level of SchemaReviewRuleLevel.
+	SchemaRuleLevelError SchemaReviewRuleLevel = "ERROR"
+	// SchemaRuleLevelWarning is the warning level of SchemaReviewRuleLevel.
+	SchemaRuleLevelWarning SchemaReviewRuleLevel = "WARNING"
+	// SchemaRuleLevelDisabled is the disabled level of SchemaReviewRuleLevel.
+	SchemaRuleLevelDisabled SchemaReviewRuleLevel = "DISABLED"
+
+	// SchemaRuleMySQLEngine require InnoDB as the storage engine.
+	SchemaRuleMySQLEngine SchemaReviewRuleType = "engine.mysql.use-innodb"
+
+	// SchemaRuleTableRequirePK require the table to have a primary key.
+	SchemaRuleTableRequirePK SchemaReviewRuleType = "table.require-pk"
+
+	// SchemaRuleTableNaming enforce the table name format.
+	SchemaRuleTableNaming SchemaReviewRuleType = "naming.table"
+
+	// SchemaRuleColumnNaming enforce the column name format
+	SchemaRuleColumnNaming SchemaReviewRuleType = "naming.column"
+
+	// SchemaRulePKNaming enforce the primary key name formaty.
+	SchemaRulePKNaming SchemaReviewRuleType = "naming.index.pk"
+
+	// SchemaRuleUKNaming enforce the unique key name format.
+	SchemaRuleUKNaming SchemaReviewRuleType = "naming.index.uk"
+
+	// SchemaRuleIDXNaming enforce the index name format.
+	SchemaRuleIDXNaming SchemaReviewRuleType = "naming.index.idx"
+
+	// SchemaRuleRequiredColumn enforce the required columns in each table.
+	SchemaRuleRequiredColumn SchemaReviewRuleType = "column.required"
+
+	// SchemaRuleColumnNotNull enforce the columns cannot have NULL value.
+	SchemaRuleColumnNotNull SchemaReviewRuleType = "column.no-null"
+
+	// SchemaRuleQueryNoSelectAll disallow 'SELECT *'.
+	SchemaRuleQueryNoSelectAll SchemaReviewRuleType = "query.select.no-select-all"
+
+	// SchemaRuleQueryRequireWhere require 'WHERE' clause.
+	SchemaRuleQueryRequireWhere SchemaReviewRuleType = "query.where.require"
+
+	// SchemaRuleQueryNoLeadingWildcardLike disallow leading '%' in LIKE, e.g. LIKE foo = '%x' is not allowed.
+	SchemaRuleQueryNoLeadingWildcardLike SchemaReviewRuleType = "query.where.no-leading-wildcard-like"
+)
+
+// NamingRulePayload is the payload for naming rule.
+type NamingRulePayload struct {
+	Format string `json:"format"`
+}
+
+// RequiredColumnRulePayload is the payload for required column rule.
+type RequiredColumnRulePayload struct {
+	ColumnList []string `json:"columnList"`
+}
+
+// SchemaReviewPolicy is the policy configuration for schema review.
+type SchemaReviewPolicy struct {
+	Name     string              `json:"name"`
+	RuleList []*SchemaReviewRule `json:"ruleList"`
+}
+
+// SchemaReviewRule is the rule for schema review policy.
+type SchemaReviewRule struct {
+	Type    SchemaReviewRuleType  `json:"type"`
+	Level   SchemaReviewRuleLevel `json:"level"`
+	Payload *string               `json:"payload"`
```

Comment:
We can use empty value for type without payload. Otherwise, it's easier to trigger NPE using pointer.